Flaky-test: PulsarStateTest.testSinkState - wrong number of messages received

```
  -     PulsarStateTest.testSinkState
         -     which gave: PulsarStateTest.testSinkState:183 expected [val1-9] but found [val1-8]
```

**To Reproduce**
I reproduced this exact issue locally by running the tests in PulsarStateTest while running:
`stress-ng --matrix 0 -t 15m`
which was running during the entire test process. 

Originally mentioned here: https://github.com/apache/pulsar/issues/6137

I fixed this issue in my PR #6202. However, now there's a new issue. 
The Github CI output just looks like this: 

`1211[ERROR] Run 4: PulsarStateTest.testSinkState:162 » PulsarAdmin org.apache.pulsar.shade.javax....`

I've attached a log with the test failure.

[11_run integration tests.txt](https://github.com/apache/pulsar/files/4207875/11_run.integration.tests.txt)

--- a/tests/integration/src/test/java/org/apache/pulsar/tests/integration/functions/PulsarStateTest.java
+++ b/tests/integration/src/test/java/org/apache/pulsar/tests/integration/functions/PulsarStateTest.java
@@ -38,6 +38,7 @@ import org.apache.pulsar.tests.integration.functions.utils.CommandGenerator;
 import org.apache.pulsar.tests.integration.functions.utils.CommandGenerator.Runtime;
 import org.apache.pulsar.tests.integration.suites.PulsarStandaloneTestSuite;
 import org.apache.pulsar.tests.integration.topologies.PulsarCluster;
+import org.awaitility.Awaitility;
 import org.testng.annotations.Test;
 
 import static java.nio.charset.StandardCharsets.UTF_8;
@@ -108,24 +109,22 @@ public class PulsarStateTest extends PulsarStandaloneTestSuite {
 
         try (PulsarAdmin admin = PulsarAdmin.builder().serviceHttpUrl(container.getHttpServiceUrl()).build()) {
 
-            retryStrategically((test) -> {
-                try {
-                    SourceStatus status = admin.sources().getSourceStatus("public", "default", sourceName);
-                    return status.getInstances().size() > 0 && status.getInstances().get(0).getStatus().numWritten > 0;
-                } catch (PulsarAdminException e) {
-                    return false;
-                }
-            }, 10, 200);
+            Awaitility.await().ignoreExceptions().untilAsserted(() -> {
+                SourceStatus status = admin.sources().getSourceStatus("public", "default", sourceName);
+                assertEquals(status.getInstances().size(), 1);
+                assertTrue(status.getInstances().get(0).getStatus().numWritten > 0);
+            });
 
-            SourceStatus status = admin.sources().getSourceStatus("public", "default", sourceName);
-            assertEquals(status.getInstances().size(), 1);
-            assertTrue(status.getInstances().get(0).getStatus().numWritten > 0);
+            {
+                FunctionState functionState =
+                        admin.functions().getFunctionState("public", "default", sourceName, "initial");
+                assertEquals(functionState.getStringValue(), "val1");
+            }
 
-            FunctionState functionState = admin.functions().getFunctionState("public", "default", sourceName, "initial");
-            assertEquals(functionState.getStringValue(), "val1");
-
-            functionState = admin.functions().getFunctionState("public", "default", sourceName, "now");
-            assertTrue(functionState.getStringValue().matches("val1-.*"));
+            Awaitility.await().ignoreExceptions().untilAsserted(() -> {
+                FunctionState functionState = admin.functions().getFunctionState("public", "default", sourceName, "now");
+                assertTrue(functionState.getStringValue().matches("val1-.*"));
+            });
         }
 
         // delete source
@@ -158,28 +157,26 @@ public class PulsarStateTest extends PulsarStandaloneTestSuite {
                     .topic(inputTopicName)
                     .create();
 
-            FunctionState functionState = admin.functions().getFunctionState("public", "default", sinkName, "initial");
-            assertEquals(functionState.getStringValue(), "val1");
+            {
+                FunctionState functionState =
+                        admin.functions().getFunctionState("public", "default", sinkName, "initial");
+                assertEquals(functionState.getStringValue(), "val1");
+            }
 
             for (int i = 0; i < numMessages; i++) {
                 producer.send("foo");
             }
 
-            retryStrategically((test) -> {
-                try {
-                    SinkStatus status = admin.sinks().getSinkStatus("public", "default", sinkName);
-                    return status.getInstances().size() > 0 && status.getInstances().get(0).getStatus().numWrittenToSink > 0;
-                } catch (PulsarAdminException e) {
-                    return false;
-                }
-            }, 10, 200);
+            Awaitility.await().ignoreExceptions().untilAsserted(() -> {
+                SinkStatus status = admin.sinks().getSinkStatus("public", "default", sinkName);
+                assertEquals(status.getInstances().size(), 1);
+                assertTrue(status.getInstances().get(0).getStatus().numWrittenToSink > 0);
+            });
 
-            SinkStatus status = admin.sinks().getSinkStatus("public", "default", sinkName);
-            assertEquals(status.getInstances().size(), 1);
-            assertTrue(status.getInstances().get(0).getStatus().numWrittenToSink > 0);
-
-            functionState = admin.functions().getFunctionState("public", "default", sinkName, "now");
-            assertEquals(functionState.getStringValue(), String.format("val1-%d", numMessages - 1));
+            Awaitility.await().ignoreExceptions().untilAsserted(() -> {
+                FunctionState functionState = admin.functions().getFunctionState("public", "default", sinkName, "now");
+                assertEquals(functionState.getStringValue(), String.format("val1-%d", numMessages - 1));
+            });
         }
 
         // delete source

